src/current/v24.1/cost-based-optimizer.md

```diff
@@ -365,12 +366,14 @@ For a join hint example, see [Use the right join type]({% link {{ page.version.v
 
 - `LOOKUP`: Forces a lookup join into the right side; the right side must be a table with a suitable index. Note that `LOOKUP` can only be used with `INNER` and `LEFT` joins.
 
-- `INVERTED`:  Forces an inverted join into the right side; the right side must be a table with a suitable [GIN index]({% link {{ page.version.version }}/inverted-indexes.md %}). Note that `INVERTED` can only be used with `INNER` and `LEFT` joins.
+- `INVERTED`: Forces an inverted join into the right side; the right side must be a table with a suitable [GIN index]({% link {{ page.version.version }}/inverted-indexes.md %}). Note that `INVERTED` can only be used with `INNER` and `LEFT` joins.
 
     {{site.data.alerts.callout_info}}
     You cannot use inverted joins on [partial GIN indexes]({% link {{ page.version.version }}/inverted-indexes.md %}#partial-gin-indexes).
     {{site.data.alerts.end}}
 
+- `STRAIGHT`: Forces a straight `INNER`, `LEFT`, or `RIGHT` join in the order specified in the query. This is used to hint the join order without hinting a join algorithm, but can potentially override a more efficient query plan.
```

Comment:
For me, it helps to think about straight joins this way:

- If `STRAIGHT JOIN` turns into a hash join or a cross join, we'll always build the hash table out of, and probe, the right side.
- If `STRAIGHT JOIN` turns into a lookup join, we'll always probe an index on the right side columns.
- If `STRAIGHT JOIN` turns into an inverted join, we'll always probe an inverted index on the right side columns.
- If `STRAIGHT JOIN` turns into a merge join nothing will be different.
- 🎵 "...we don't talk about zigzag joins, no no! we don't talk about about zig zag..." 🎵

And confusingly this is all independent of whether the join is an INNER, LEFT, or RIGHT join. (Though I don't think `RIGHT STRAIGHT JOIN` can turn into a lookup or an inverted join.)

Another way to think of it is that straight join is always streaming with regards to the left side.

Reply:
@michae2 Thanks for the clarifications! Can you help me understand what "probe" and "streaming [with regards to the left side]" mean in this context? Those terms aren't used in this doc, so I wonder if there is another way I can phrase that information, or if the meaning would be self-evident to the reader.

Reply:
Ah, sorry. "Probe" means "look into" i.e. find a hash value in a hash table. You can see it used in this article for example: https://en.wikipedia.org/wiki/Hash_join

"Streaming" means "begins producing results before all of the input is consumed". Normally we would say hash join is "blocking" rather than "streaming" because it has to consume the entire right side input before producing output, but by "streaming with regards to the left side" I mean that once the hash table is built it can begin producing output as soon as it has read some of the left input.

Not sure if any of this needs to go into docs... just how I think about it.